### PR TITLE
Fetched, Rendered, and Filtered rockets

### LINKED ISSUE #3: [2pt]  Display reserved rockets in My profile - Filters
Render a list of all reserved rockets (use `filter()`) on the "My profile" page.

### LINKED ISSUE #5: [1pt] Switch badges for Rockets - Conditional components
Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design)

Use the React conditional rendering syntax:
```javascript
{rocket.reserved && ( 
    // render Cancel Rocket button
)}
```

### LINKED ISSUE #12: [3pt] Implement rocket booking - Actions
- When a user clicks the "Reserve rocket" button, action needs to be dispatched to update the store. You need to get the ID of the reserved rocket and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all rockets, but the selected rocket will have an extra key `reserved` with its value set to `true`. You could use a JS `filter()` or `map()` to set the value of the new state - i.e.:
```javascript
const newState = state.map(rocket => {
    if(rocket.id !== id) 
        return rocket;
    return { ...rocket, reserved: true };
});

```

- Regardless of which method you choose, make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct rocket ID as an argument.

### LINKED ISSUE #13: [4pt] Display rockets - Lists render
- Use `useSelector()` Redux Hook to select the state slices and render lists of rockets in corresponding routes. i.e.:
```javascript
// get rockets data from the store
const rockets = useSelector(state => state.rockets);
```

- You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
- Render a list of rockets (as per design). For the image of a rocket use the first image in the array of `flickr_images`.

### LINKED ISSUE #17: [4pt]  Fetch rockets - Fetch data
Fetch data from the Rockets endpoint  (https://api.spacexdata.com/v3/rockets) when the application starts (as Rockets is the default view).

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- id
- name
- type
- flickr_images

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### LINKED ISSUE #21: [1pt] Create basic structure for Rockets - Setup
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create a directory for all Redux state slice files.

### PR DESCRIPTION
**Pull request by Shakir-Hussain12**
**For this PR, I :**
- Created a basic structure for `Rockets` section. (Closes #21)
- Fetched Rockets data from [API](https://api.spacexdata.com/v4/rockets). (Closes #17)
- Displayed requested rocket data. (Closes #13)
- Added actions to `Add` and `Cancel` Rocket Booking. (Closes #12, #8)
- Added Badges to `Booked` rockets. (Closes #5)
- Displayed reserved Rockets on `My Profile` section. (Closes #3)
- Will add improved styles later.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -16,10 +18,15 @@ function App() {
...
 
+  useEffect(() => {
+    dispatch(getRockets());
+  }, [dispatch]);
+
   return (
     <Routes>
       <Route path="/" element={<Layout />}>
         <Route index element={<Missions />} />
+        <Route path="/rockets" element={<Rockets />} />
```
Comment (Timbar09) [inline src/App.jsx]: I think you should replace the Missions component and link with the Rockest component and link as the home page in order to align with the setup of the medium-fidelity wireframes.

- The medium-fidelity wireframe links
links

- Our page
Shakir links

**Review by Timbar09 — CHANGES_REQUESTED** [review]
**Changes Required** ♻️ 
---
Hi @Shakir-Hussain12, Great job so far, but there is just one change you need to make.

**Positives**
- The functionality is good
- Nice simple design

**Changes**
- Check the inline comment above.

File: src/App.jsx (hunk 1 above)
Comment (Shakir-Hussain12) [reply src/App.jsx]: Thanks, @Timbar09, for the suggestion. I will make these changes ASAP.

**Review by Timbar09 — APPROVED** [review]
**Approved** ✅ 
---
Hi @Shakir-Hussain12, Great job applying the changes.

It's safe to merge now.

**Pull request merged**